src/test/java/com/powsybl/ws/commons/ComputationTest.java

```diff
@@ -0,0 +1,268 @@
+import com.powsybl.ws.commons.computation.service.AbstractComputationRunContext;
+import com.powsybl.ws.commons.computation.service.AbstractComputationService;
+import com.powsybl.ws.commons.computation.service.AbstractResultContext;
+import com.powsybl.ws.commons.computation.service.AbstractWorkerService;
+import com.powsybl.ws.commons.computation.service.ExecutionService;
+import com.powsybl.ws.commons.computation.service.NotificationService;
+import com.powsybl.ws.commons.computation.service.ReportService;
+import com.powsybl.ws.commons.computation.service.UuidGeneratorService;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.simple.SimpleMeterRegistry;
+import io.micrometer.observation.ObservationRegistry;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import org.assertj.core.api.WithAssertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.support.MessageBuilder;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import static com.powsybl.ws.commons.ComputationTest.MockComputationStatus.COMPLETED;
+import static com.powsybl.ws.commons.computation.service.NotificationService.HEADER_RECEIVER;
+import static com.powsybl.ws.commons.computation.service.NotificationService.HEADER_RESULT_UUID;
+import static com.powsybl.ws.commons.computation.service.NotificationService.HEADER_USER_ID;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith({ MockitoExtension.class })
+@Slf4j
+class ComputationTest implements WithAssertions {
+    private static final String COMPUTATION_TYPE = "mockComputation";
+    @Mock
+    private NetworkStoreService networkStoreService;
+    @Mock
+    private ReportService reportService;
+    @Mock
+    private ExecutionService executionService;
+    @Mock
+    private NotificationService notificationService;
+    @Mock
+    private ObjectMapper objectMapper;
+    @Mock
+    private Network network;
+
+    public static class MockComputationResult { }
+
+    public static class MockComputationParameters { }
+
+    public enum MockComputationStatus {
+        NOT_DONE,
+        RUNNING,
+        COMPLETED
+    }
+
+    @Service
+    public static class MockComputationResultService extends AbstractComputationResultService<MockComputationStatus> {
```

Comment:
In my opinion, using Mockito.mock() is more flexible if we want to test more different scenarios in the future..

Reply:
OK we keep concrete classes in test